src/cert_store.rs

```diff
@@ -63,11 +72,145 @@ impl CertStore {
+                                               winapi::CERT_SYSTEM_STORE_CURRENT_USER,
+                                               data.as_ptr() as *mut _);
+            if store.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(store))
+            }
+        }
+    }
+
+    /// Opens up the specified key store within the context of the local
+    /// machine.
+    ///
+    /// Known valid values for `which` are "Root" and "My".
+    pub fn open_local_machine(which: &str) -> io::Result<CertStore> {
+        unsafe {
+            let data = OsStr::new(which)
+                             .encode_wide()
+                             .chain(Some(0))
+                             .collect::<Vec<_>>();
+            let store = crypt32::CertOpenStore(winapi::CERT_STORE_PROV_SYSTEM_W as winapi::LPCSTR,
+                                               0,
+                                               0,
+                                               winapi::CERT_SYSTEM_STORE_LOCAL_MACHINE,
+                                               data.as_ptr() as *mut _);
+            if store.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(store))
+            }
+        }
+    }
+
+    /// Imports a PKCS#12-encoded key/certificate pair, returned as a
+    /// `CertStore` instance.
+    ///
+    /// The password must also be provided to decrypt the encoded data.
+    pub fn import_pkcs12(data: &[u8], password: &str) -> io::Result<CertStore> {
+        unsafe {
+            let mut blob = winapi::CRYPT_INTEGER_BLOB {
+                cbData: data.len() as winapi::DWORD,
+                pbData: data.as_ptr() as *mut u8,
+            };
+            let password = OsStr::new(password)
+                                 .encode_wide()
+                                 .chain(Some(0))
+                                 .collect::<Vec<_>>();
+            let res = crypt32::PFXImportCertStore(&mut blob,
+                                                  password.as_ptr(),
+                                                  0);
+            if res.is_null() {
+                Err(io::Error::last_os_error())
+            } else {
+                Ok(CertStore(res))
+            }
+        }
+    }
+
+    /// Returns an iterator over the certificates in this certificate store.
+    pub fn certs(&mut self) -> Certs {
+        Certs { store: self, cur: None }
+    }
+
+    /// Adds a certificate context to this store.
+    ///
+    /// This function will add the certificate specified in `cx` to this store.
+    /// A copy of the added certificate is returned.
+    pub fn add_cert(&mut self, cx: &CertContext) -> io::Result<CertContext> {
+        unsafe {
+            let flags = winapi::CERT_STORE_ADD_REPLACE_EXISTING;
+            let mut ret = ptr::null();
+            let res = crypt32::CertAddCertificateContextToStore(self.0,
```

Comment:
I think we'd probably want to expose this a bit more powerfully (e.g. the different replacement strategies)